code/game/objects/structures/bounty_post.dm

```diff
@@ -0,0 +1,268 @@
+	qdel(src)
+
+/datum/holy_bounty/proc/create_bounty()
+	var/list/possible_turfs = get_area_turfs(/area/icemoon/underground/unexplored/rivers)
+	var/turf/chosen_turf
+	shuffle_inplace(possible_turfs)
+
+	for(var/turf/possible_turf as anything in possible_turfs)
+		if(!possible_turf.is_blocked_turf() && !isgroundlessturf(possible_turf))
+			chosen_turf = possible_turf
+			break
+
+	if(isnull(chosen_turf))
+		return null
+
+	var/atom/created_bounty = new bounty_typepath(chosen_turf)
+	gps_location = "Bounty [rand(1,1000)]"
+	created_bounty.AddComponent(/datum/component/gps, gps_location)
+	return created_bounty
+
+/datum/holy_bounty/Destroy()
+	. = ..()
+	GLOB.holy_contracts -= src
+	assignee = null
+
+/////bounty contract
+/obj/item/bounty_contract
+	name = "bounty contract"
+	icon = 'icons/obj/scrolls.dmi'
+	icon_state = "bounty_paper"
+	item_flags = NOBLUDGEON
+	///our bounty
+	var/datum/holy_bounty/our_bounty
+
+/obj/item/bounty_contract/proc/set_bounty(datum/bounty)
+	our_bounty = bounty
+	RegisterSignal(our_bounty, COMSIG_QDELETING, PROC_REF(solve_contract))
+
+/obj/item/bounty_contract/ui_interact(mob/user, datum/tgui/ui)
+	if(isnull(our_bounty))
+		return
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "bountypaper")
+		ui.open()
+
+/obj/item/bounty_contract/ui_static_data(mob/user)
+	var/list/data = list()
+	data["bounty_name"] = our_bounty.bounty_name
+	data["bounty_icon"] = icon2base64(our_bounty.bounty_icon)
+	data["bounty_reward"] = our_bounty.reward_points
+	data["bounty_gps"] = our_bounty.gps_location
+	return data
+
+/obj/item/bounty_contract/proc/solve_contract(datum/source)
+	SIGNAL_HANDLER
+	our_bounty = null
+
+////our bounty billboard
+/obj/structure/bounty_post
+	name = "Bounty billboard"
+	desc = "An honest man's work!"
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "bounty_post"
+	density = TRUE
+	anchored = TRUE
+	resistance_flags = INDESTRUCTIBLE
+	///maximum amount of bounties we can hold
+	var/bounty_amount = 8
+
+/obj/structure/bounty_post/Initialize(mapload)
+	. = ..()
+	setup_bounties()
+
+/obj/structure/bounty_post/proc/setup_bounties()
+	var/bounties_to_set = bounty_amount - length(GLOB.holy_contracts)
+	for(var/count in 1 to bounties_to_set)
+		new /datum/holy_bounty
+	addtimer(CALLBACK(src, PROC_REF(setup_bounties)), 10 MINUTES) //every 10 minutes, regenerate bounties
+
+/obj/structure/bounty_post/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "bountypost")
+		ui.open()
+
+/obj/structure/bounty_post/ui_data(mob/user)
+	var/list/data = list()
+	data["bounty"] = list()
+	for(var/datum/holy_bounty/our_bounty as anything in GLOB.holy_contracts)
+		if(!isnull(our_bounty.assignee?.resolve()))
+			continue //dont display bounties that are already claimed
+		data["bounty"] += list(list(
+			"bounty_id" = REF(our_bounty),
+			"bounty_name" = our_bounty.bounty_name,
+			"bounty_icon" = icon2base64(our_bounty.bounty_icon),
```

Comment:
You should use DMicon rather than icon2base64, it will likely look better, perform better, and be easier to work with

All you need to do is pass `icon` and `icon_state`